src/shogun/base/SGObject.h

```diff
@@ -784,6 +766,38 @@ class SGObject: public std::enable_shared_from_this<SGObject>
 	}
 
 protected:
+	template<typename T>
+	void register_parameter_visitor() const
+	{
+		Any::register_visitor<T, ParameterPutInterface<T>>(
+			[](T* value, auto* visitor)
+			{
+				*value = visitor->m_value;	
+			}
+		);
+		if constexpr (traits::is_functional<T>::value)
+		{
+			if constexpr (!traits::returns_void<T>::value)
+			{
+				using ReturnType = typename T::result_type;
+				Any::register_visitor<T, ParameterGetterInterface<ReturnType, T>>(
+					[](T* value, auto* visitor)
+					{
+						visitor->m_value = value->operator()();
+					}
+				);
+			}
+		}
+		else
+		{
+			Any::register_visitor<T, ParameterGetterInterface<T, T>>(
+				[](T* value, auto* visitor)
+				{
+					visitor->m_value = *value;	
+				}
+			);
+		}
+	}
```

Comment:
@vigsterkr I adapted from your PR. Can then put all the visitor registrations in this function

Reply:
👍